src/Symfony/Component/Uri/Tests/UriTest.php

```diff
@@ -0,0 +1,253 @@
+
+    public function testUriWithFragmentTextDirectives()
+    {
+        $uri = Uri::parse('https://example.com');
+        $newUri = $uri->withFragmentTextDirective('start', 'end', 'prefix', 'suffix');
+
+        $this->assertNull($uri->fragmentTextDirective);
+        $this->assertNotNull($newUri->fragmentTextDirective);
+
+        $this->assertNotSame($uri, $newUri);
+
+        $this->assertSame('https://example.com#:~:text=prefix-,start,end,-suffix', (string) $newUri);
+    }
+
+    public function testUriWithExistingFragmentWithFragmentTextDirectives()
+    {
+        $uri = Uri::parse('https://example.com#existing');
+        $newUri = $uri->withFragmentTextDirective('start', 'end', 'prefix', 'suffix');
+
+        $this->assertNull($uri->fragmentTextDirective);
+        $this->assertNotNull($newUri->fragmentTextDirective);
+
+        $this->assertNotSame($uri, $newUri);
+
+        $this->assertSame('https://example.com#existing:~:text=prefix-,start,end,-suffix', (string) $newUri);
+    }
+
+    public function testUriWithIdnHostAsAscii()
+    {
+        $uri = Uri::parse('https://bücher.ch');
+        $this->assertSame('bücher.ch', $uri->host);
+
+        $uri = $uri->withIdnHostAsAscii();
+        $this->assertSame('xn--bcher-kva.ch', $uri->host);
+    }
+
+    public function testUriWithIdnHostAsUnicode()
+    {
+        $uri = Uri::parse('https://xn--bcher-kva.ch');
+        $this->assertSame('xn--bcher-kva.ch', $uri->host);
+
+        $uri = $uri->withIdnHostAsUnicode();
+        $this->assertSame('bücher.ch', $uri->host);
+    }
+
+    /**
+     * @dataProvider provideResolveUri
+     */
+    public function testResolveUri(string $baseUri, string $relativeUri, string $expectedUri)
+    {
+        $resolvedUri = Uri::resolve($baseUri, $relativeUri);
+
+        $this->assertSame($expectedUri, $resolvedUri);
+    }
+
+    public static function provideResolveUri(): iterable
+    {
+        yield 'Single-level absolute path' => ['https://example.com/foo', '/bar', 'https://example.com/bar'];
+        yield 'Multi-level Absolute path' => ['https://example.com', '/foo/bar', 'https://example.com/foo/bar'];
+        yield 'Single-level relative path' => ['https://example.com/foo', 'bar', 'https://example.com/foo/bar'];
+        yield 'Single-level relative path with trailing slash' => ['https://example.com/foo/', 'bar', 'https://example.com/foo/bar'];
+        yield 'Single-level parent' => ['https://example.com/foo/', '../bar', 'https://example.com/bar'];
+        yield 'Multi-level parent' => ['https://example.com/foo/', '../../bar', 'https://example.com/bar'];
+        yield 'Absolute path with trailing slash' => ['https://example.com/foo/', '/bar', 'https://example.com/bar'];
+        yield 'Different domains' => ['https://example.com/foo/', 'https://example.org/bar', 'https://example.org/bar'];
+        yield 'Keep query string and fragment' => ['https://example.com/foo/?foo=bar#baz', '/bar', 'https://example.com/bar?foo=bar#baz'];
```

Comment:
I may have misunderstood this, but it seems to me everything after the path should be lost while resolving the relative URI.

https://www.rfc-editor.org/rfc/rfc3986#page-36

Reply:
Fixed as well in https://github.com/symfony/symfony/compare/33b04577f0d2ef8b2973f81fae92ae2406583d1a..9f9184830d485f1571f6a836676329d055844731 🙂